server/build_event_protocol/build_event_handler/build_event_handler.go

```diff
@@ -287,11 +287,50 @@ func (r *statsRecorder) flushInvocationStatsToOLAPDB(ctx context.Context, ij *in
 	}
 
 	err = r.env.GetOLAPDBHandle().FlushInvocationStats(ctx, inv)
+	if err != nil {
+		return err
+	}
 	// Temporary logging for debugging clickhouse missing data.
-	if err == nil {
-		log.CtxInfo(ctx, "successfully wrote invocation to clickhouse")
+	log.CtxInfo(ctx, "Successfully wrote invocation to clickhouse")
+
+	if r.env.GetExecutionCollector() == nil {
+		return nil
 	}
-	return err
+	const batchSize = 500_000
```

Comment:
Do we need such a large batch size? Would for example 50k work just as well and reduce potential for large memory spikes?